### PR TITLE
Allow access to a site to be revoked

### PR DESCRIPTION
**Pull request by grs**
Provides a CLI command with which to revoke access for any inbound links to a site, by regenerating the CA used to issue certificates for TLS connections.

### CODE REVIEW

File: client/revoke_all.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	ca, err := kube.RegenerateCertAuthority(types.SiteCaSecret, cli.Namespace, cli.KubeClient)
+	if err != nil {
+		return err
+	}
+	siteServerSecret := types.Credential{
+		Name:    types.SiteServerSecret,
+		Subject: types.TransportServiceName,
+	}
+	usingRoutes := true
+	if cli.RouteClient != nil {
```
Comment (fgiorgetti) [inline client/revoke_all.go:34]: If `RouteClient` is nil, I guess you have to update `usingRoutes` to false as well.

File: client/revoke_all.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+			usingRoutes = false
+		} else if err != nil {
+			return err
+		}
+		siteServerSecret.Hosts = append(siteServerSecret.Hosts, route.Spec.Host)
+		route, err = kube.GetRoute(types.EdgeRouteName, cli.Namespace, cli.RouteClient)
+		if err != nil {
+			return err
+		}
+		siteServerSecret.Hosts = append(siteServerSecret.Hosts, route.Spec.Host)
```
Comment (fgiorgetti) [inline client/revoke_all.go:46]: If `usingRoutes` is false at this point, you won't be able to get `route.Spec.Host`.
Maybe all this block could be part of an else statement.

File: cmd/skupper/skupper.go [hunk 3]
```diff
@@ -882,6 +882,26 @@ func NewCmdDebugDump(newClient cobraFunc) *cobra.Command {
 	return cmd
 }
 
+func NewCmdRevokeaccess(newClient cobraFunc) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "revoke-access",
+		Short: "Revoke all previously granted acces to the site the site.",
```
Comment (fgiorgetti) [inline cmd/skupper/skupper.go:888]: typos: `acces` and duplicated `the site`.

**Review by fgiorgetti — COMMENTED** [review]
A few comments reviewing the code changes. I will play with it using
the other PR that includes claims support and provide further feedback.

**Comment by grs** [comment]
Thanks @fgiorgetti ! I have made the same changes in the other PR. Probably simplest now to wait for claims to land then merge that one when everyone is happy. If you agree I'll close this one now.

**Comment by fgiorgetti** [comment]
> Thanks @fgiorgetti ! I have made the same changes in the other PR. Probably simplest now to wait for claims to land then merge that one when everyone is happy. If you agree I'll close this one now.

Sure thing. Thank you!

**Pull request closed without merging**